dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java

```diff
@@ -565,35 +537,73 @@ private static String getServiceDescriptorDirectory() {
        }
 
        if(UtilMethods.isNotSet(appDescriptor.getName())){
-           throw new DotDataException("The required field `name` isn't set on the incoming file.");
+           throw new DotDataValidationException("The required field `name` isn't set on the incoming file.");
        }
 
        if(UtilMethods.isNotSet(appDescriptor.getDescription())){
-           throw new DotDataException("The required field `description` isn't set on the incoming file.");
+           throw new DotDataValidationException("The required field `description` isn't set on the incoming file.");
        }
 
        if(UtilMethods.isNotSet(appDescriptor.getIconUrl())){
-           throw new DotDataException("The required field `iconUrl` isn't set on the incoming file.");
+           throw new DotDataValidationException("The required field `iconUrl` isn't set on the incoming file.");
        }
 
        if(!UtilMethods.isSet(appDescriptor.getParams())){
-           throw new DotDataException("The required field `params` isn't set on the incoming file.");
+           throw new DotDataValidationException("The required field `params` isn't set on the incoming file.");
+       }
+
+       for (Map.Entry<String, ParamDescriptor> entry : appDescriptor.getParams().entrySet()) {
+           validateParamDescriptor(entry.getKey(), entry.getValue());
        }
 
        return true;
 
    }
 
+   private void validateParamDescriptor(final String name, final ParamDescriptor descriptor) throws DotDataValidationException {
+
+       if(UtilMethods.isNotSet(name)){
+           throw new DotDataValidationException("Param descriptor is missing required  field `name` .");
+       }
+
+       if(name.length() > 100){
+           throw new DotDataValidationException(String.format("Param name `%s` exceeds 100 chars length.", name));
+       }
+
+       if(UtilMethods.isNotSet(descriptor.getHint())){
+           throw new DotDataValidationException("Param descriptor `%s` is missing required field `hint` .");
+       }
+
+       if(UtilMethods.isNotSet(descriptor.getLabel())){
+           throw new DotDataValidationException(String.format("Param descriptor `%s` is missing required field `hint` .",name));
+       }
+
+       if(null == descriptor.getType()){
+           throw new DotDataValidationException(String.format("Param descriptor `%s` is missing required field `type` .",name));
+       }
+
+       if (Type.BOOL.equals(descriptor.getType()) && descriptor.isHidden() ) {
+           throw new DotDataValidationException(String.format("Boolean Params like `%s` can not be marked hidden.", name));
+       }
+
+       if ("null".equalsIgnoreCase(descriptor.getValue()) && descriptor.isRequired()) {
+           throw new DotDataValidationException(String.format(
+                   "Null isn't allowed as the default value on required params see `%s`. ",
+                   name)
+           );
+       }
+   }
```

Comment:
are you sure that you are testing all this cases?

Reply:
Tests added